src/model/transaction/addEntityToContentState.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright (c) 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule addEntityToContentState
+ * @typechecks
+ * @flow
+ */
+
+'use strict';
+
+const generateRandomKey = require('generateRandomKey');
+
+import type ContentState from 'ContentState';
+import type DraftEntityInstance from 'DraftEntityInstance';
+
+function addEntityToContentState(
+  contentState: ContentState,
+  instance: DraftEntityInstance
+): ContentState {
+  const newKey = generateRandomKey();
```

Comment:
I think it would be okay to keep using a regular incrementing counter for these IDs. The random key generator has mostly been useful for being able to spot a unique key within the DOM, etc., while here it's fine for it to just be a number.